crates/dbsp/src/operator/group/topk.rs

```diff
@@ -378,10 +382,14 @@ where
         let mut count = 0usize;
 
         while cursor.key_valid() && count < self.k {
-            let w = cursor.weight();
-            if !w.is_zero() {
+            let mut w = cursor.weight();
+            while w.ge0() && !w.is_zero() {
```

Comment:
Better hope this is not negative

Reply:
If it's negative, it'll just skip the key, which seems like the most sensible thing to do here?